src/v/cluster/scheduling/partition_allocator.cc

```diff
@@ -72,6 +75,12 @@ partition_allocator::allocate_partition(partition_constraints p_constraints) {
         effective_constraits.hard_constraints.push_back(
           ss::make_lw_shared<hard_constraint_evaluator>(
             distinct_from(replicas.get())));
+
+        // rack-placement contraint
+        effective_constraits.soft_constraints.push_back(
```

Comment:
Do we want this constraint to be always enabled ? Should we make it configurable ?

Reply:
Makes sense. I added the option.